scene/main/node.cpp

```diff
@@ -2591,7 +2591,12 @@ String Node::to_string() {
 			return ret;
 		}
 	}
-
+	if (_get_extension() && _get_extension()->to_string) {
+		String ret;
+		GDExtensionBool is_valid;
+		_get_extension()->to_string(_get_extension_instance(), &is_valid, &ret);
+		return ret;
+	}
```

Comment:
`is_valid` has neither a defined input value (uninitialized), nor is its output checked -- is that intended?

Reply:
Those are good questions! However, this code exactly matches what we already have in `Object::to_string()`, so it isn't making things any worse. If we do end up evaluating how `is_valid` should be used, then it should be fixed in both places.

Reply:
Yeah, I just copied `Object::to_string` but I also thought it was weird that `is_valid` is unused. We should probably check before returning, like we do with `valid` in scripts:

https://github.com/godotengine/godot/blob/e96ad5af98547df71b50c4c4695ac348638113e0/core/object/object.cpp#L926-L932

Note that `valid` is also uninitialized. I guess this variable is always assigned by `to_string` so there's no need to initialize it, but I'm not too familiar with C++ so if we should always initialize it just let me know.

Reply:
At least the default implementations of `to_string()` seem to properly set `r_valid` to false before errors, and `true` on success.

That's a common pattern for calling methods on ScriptInstance or GDExtension, though it's hard to guarantee that it's always safe. The tradeoff between avoiding one potentially unneeded initialization and risking using uninitialized memory needs to be assessed (especially whether _not_ initializing these actually makes a difference to performance at all).

It's a much bigger scope than just this PR though. For this I'd suggest just putting a comment in both implementations that they should be kept in sync (not for what to do with `valid` specifically, but the whole method).